### PR TITLE
Implement Battlemetrics API

### PR DESCRIPTION
**Pull request by FaiThiX**
i have changed the scrape process from a website based data retrieval to the api so that you don't get an error because the bot didn't get access to the website. if you need any changes to make the merge work, post it below. I have tested everything and hope that I have not overlooked a bug. if you have any questions, please write here or in discord to 'Rene | FaiThiX#0420'.

### CODE REVIEW

File: src/discordTools/discordEmbeds.js [hunk 1]
```diff
@@ -81,8 +81,8 @@ module.exports = {
             thumbnail: `${server.img}`,
             fields: [{
                 name: Client.client.intlGet(guildId, 'connect'),
-                value: `\`${server.connect === null ?
```
Comment (alexemanuelol) [inline src/discordTools/discordEmbeds.js:84]: Remove these changes

File: src/util/FcmListener.js [hunk 2]
```diff
@@ -228,7 +228,7 @@ async function pairingServer(client, guild, full, data, body) {
         switchGroups: server ? server.switchGroups : {},
         messageId: (message !== undefined) ? message.id : null,
         battlemetricsId: battlemetricsId,
-        connect: (info === null) ? 'Unavailable' : `${client.intlGet(guild.id, 'connect')} ${info.ip}:${info.port}`,
```
Comment (alexemanuelol) [inline src/util/FcmListener.js:231]: You can remove these changes and replace 'Unavailable' with null.

File: src/util/battlemetricsAPI.js [hunk 3]
```diff
@@ -74,19 +72,15 @@ module.exports = {
                 return null;
             }
         }
-
-        let regex = new RegExp(
-            `"id":"${serverId}".*?"ip":"(.+?)","port":(.+?),".*?"rank":(\\d+?),".*?"country":"(.+?)","status":"(.+?)"`, 'g');
-        let data = regex.exec(page);
-
+        data = page['data']['attributes']
```
Comment (alexemanuelol) [inline src/util/battlemetricsAPI.js:75]: "ReferenceError: data is not defined", just put a let at the start of the line.

File: src/util/battlemetricsAPI.js [hunk 4]
```diff
@@ -141,4 +135,15 @@ module.exports = {
...
         return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
     },
+
+    formatTime: async function (timestamp) {
+        const date = new Date(timestamp);
+        const now = Date.now();
+        const diffMs = now - date;
+        const diffHours = Math.floor(diffMs / (1000 * 60 * 60));
+        const diffMinutes = Math.floor((diffMs / (1000 * 60)) % 60);
+        const hoursStr = diffHours.toString().padStart(2, "0");
```
Comment (alexemanuelol) [inline src/util/battlemetricsAPI.js:145]: Just some OCD stuff, could you replace " with '? :D

**Comment by alexemanuelol** [comment]
Other than the comments, looks neat! Fantastic job!

**Comment by RadNotRed** [comment]
Looks good, will be much better than scraping
Glad to see that this is getting added because I was about to do it myself

File: src/util/FcmListener.js (hunk 2 above)
Comment (FaiThiX) [reply src/util/FcmListener.js]: I am not quite sure what you mean by that :/

**Comment by FaiThiX** [comment]
added your suggestions @alexemanuelol

**Pull request merged**